### PR TITLE
Add heap usage stats with defmt

### PR DESCRIPTION
**Pull request by okhsunrog**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [ ] I have updated existing examples or added new ones (if applicable).
- [x] I have used `cargo xtask fmt-packages` command to ensure that all changed code is formatted correctly.
- [x] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
- [ ] I have added necessary changes to user code to the [Migration Guide](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/MIGRATING-0.21.md).
- [x] My changes are in accordance to the [esp-rs API guidelines](https://github.com/esp-rs/esp-hal/blob/main/documentation/API-GUIDELINES.md)

#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/documentation/CONTRIBUTING.md) and followed its instructions.

### Pull Request Details 📖

This is based on https://github.com/esp-rs/esp-hal/pull/2137 with defmt::Format trait properly implemented and minor changes in format of heap stats, tested with defmt on esp32-c3

### CODE REVIEW

File: qa-test/src/bin/psram_quad.rs
Comment (MabezDev) [inline qa-test/src/bin/psram_quad.rs:5]:
```suggestion
//% CHIPS: esp32 esp32s2 esp32s3
```

**Review by MabezDev — CHANGES_REQUESTED** [review]

**Review by MabezDev — APPROVED** [review]
Thanks for taking care of this! Much appreciated.

LGTM!

File: esp-alloc/Cargo.toml
Comment (playfulFence) [inline esp-alloc/Cargo.toml:31]: Just a nitpick, but could you please align the indentation 😅 
```suggestion
defmt                 = { version = "0.3.8", optional = true }
cfg-if                = "1.0.0"
critical-section      = "1.1.3"
enumset               = "1.1.5"
linked_list_allocator = { version = "0.10.5", default-features = false, features = ["const_mut_refs"] }
document-features     = "0.2.10"
```

File: esp-alloc/src/lib.rs
Comment (playfulFence) [inline esp-alloc/src/lib.rs:55]:
```suggestion
//! You can also get stats about the heap usage at anytime with:
```

File: esp-alloc/src/lib.rs
Comment (playfulFence) [inline esp-alloc/src/lib.rs:58]: Also just a nitpick, feel free to ignore. It was written this way further down in the code, I thought it would be nice to make it the same 😄.
```suggestion
//! // HeapStats implements the Display and defmt::Format traits, so you can pretty-print the heap stats.
```

File: esp-alloc/src/lib.rs
Comment (playfulFence) [inline esp-alloc/src/lib.rs:564]: I would write it that way probably, but again, feel free to ignore!
```suggestion
                // We need to call `used()` because [linked_list_allocator::Heap] does internal size
```

**Review by playfulFence — CHANGES_REQUESTED** [review]
Спасибо! 😉❤️ 

Changes look good to me overall, just a handful of docs-related comments. As soon as it's answered/ignored - I'll approve it!

File: esp-alloc/src/lib.rs
Comment (okhsunrog) [reply esp-alloc/src/lib.rs:58]: I don't mind, let's make it more consistent :)

**Comment by okhsunrog** [comment]
@playfulFence thanks for you comments, it's indeed cleaner this way. 
Waiting for an approval :D

**Comment by playfulFence** [comment]
Hahaha 🤣 
image
I appreciate the sense of humour! 
As soon as CI passes - we'll merge it, thanks again ❤️

**Comment by playfulFence** [comment]
image

image

**Review by playfulFence — CHANGES_REQUESTED** [review]
Thanks!

edit: oops

**Pull request merged**
